recipes/cudatoolkit-dev/cudatoolkit-dev-post-install.py

```diff
@@ -0,0 +1,388 @@
+            os.makedirs(deactivate_dir_path)
+
+        except FileExistsError:
+            pass
+
+        # Copy cudatoolkit-dev-activate and cudatoolkit-dev-deactivate
+        # to activate.d and deactivate.d directories
+
+        scripts_dir = Path(self.prefix) / 'scripts'
+        activate_scripts_dir = scripts_dir / 'activate.d'
+        deactivate_scripts_dir = scripts_dir / 'deactivate.d'
+
+        activate_scripts_list = [
+            "cudatoolkit-dev-activate.sh",
+            "cudatoolkit-dev-activate.bat"]
+        for file_name in activate_scripts_list:
+            file_full_path = activate_scripts_dir / file_name
+            shutil.copy(file_full_path, activate_dir_path)
+
+        deactivate_scripts_list = [
+            "cudatoolkit-dev-deactivate.sh",
+            "cudatoolkit-dev-deactivate.bat"]
+
+        for file_name in deactivate_scripts_list:
+            file_full_path = deactivate_scripts_dir / file_name
+            shutil.copy(file_full_path, deactivate_dir_path)
+
+    def download_blobs(self):
+        """Downloads the binary blobs to the $SRC_DIR
+        """
+        dl_url = urlparse.urljoin(self.base_url, self.installers_url_ext)
+        dl_url = urlparse.urljoin(dl_url, self.cu_blob)
+        dl_path = os.path.join(self.src_dir, self.cu_blob)
+        if not self.debug_install_path:
+            print("downloading %s to %s" % (dl_url, dl_path))
+            download(dl_url, dl_path)
+
+        else:
+            existing_file = os.path.join(self.debug_install_path, self.cu_blob)
+            print("DEBUG: copying %s to %s" % (existing_file, dl_path))
+            shutil.copy(existing_file, dl_path)
+
+    def check_md5(self):
+        """Checks the md5sums of the downloaded binaries
+        """
+        md5file = self.md5_url.split("/")[-1]
+        path = os.path.join(self.src_dir, md5file)
+        download(self.md5_url, path)
+
+        # compute hash of blob
+        blob_path = os.path.join(self.src_dir, self.cu_blob)
+        md5sum = hashsum_file(blob_path, 'md5')
+
+        # get checksums
+        with open(path, 'r') as f:
+            checksums = [x.strip().split() for x in f.read().splitlines() if x]
+
+        # check md5 and filename match up
+        check_dict = {x[0]: x[1] for x in checksums}
+        assert check_dict[md5sum].startswith(self.cu_blob[:-7])
+
+    def extract(self, *args):
+        """The method to extract files from the cuda binary blobs.
+        Platform specific extractors must implement.
+        """
+        raise RuntimeError("Must implement")
+
+    def cleanup(self):
+        """The method to delete unnecessary files after
+        the installation process.
+        """
+        raise RuntimeError("Must implement")
+
+
+class WindowsExtractor(Extractor):
+    """The windows extractor
+    """
+
+    def extract(self):
+        print("Extracting on Windows.....")
+        runfile = os.path.join(self.src_dir, self.cu_blob)
+        cmd = ['7za', 'x', '-o%s' %
```

Comment:
I am not sure we can gaurantee that 7zip is installed on all machines, but since we are skipping windows, I am not sure it is a big deal.

Reply:
I had an implementation for windows extractor class before deciding to skip windows during the build process. I forgot to remove it . For now, I will remove this extractor until windows support becomes necessary.